src/cryptography/hazmat/primitives/ciphers/modes.py

```diff
@@ -54,13 +63,25 @@ def tag(self):
         """
 
 
+def _check_aes_key_length(self, algorithm):
+    if algorithm.key_size > 256 and algorithm.name == "AES":
```

Comment:
Ugh, this is confusing and seems wrong. For one, it's equally true of any other block cipher right?

Reply:
Previously when `Cipher` called `validate_for_algorithm` against the mode we assumed that the key length had already been enforced by the constructor for the algorithm. Since now the length of the key is dependent on the  mode (but only for AES, which is the only algorithm where we support a mode with such behavior) to preserve previous behavior we need to do this check. (Side note: If we're okay with having it fail when calling encryptor/decryptor then we could remove this check here since it will fail with `UnsupportedAlgorithm` when it tries to instantiate an `EVP_CIPHER *`)

Other block ciphers are not affected because we didn't touch their allowed key sizes. It is true that a user attempting to do Camellia+XTS would see an apparent inconsistency with being unable to even provide a 512 bit key to the Camellia constructor, but I don't see a way to fix that without forcing consumers to (very confusingly IMO) split the key in half.